### PR TITLE
8264173: [s390] Improve Hardware Feature Detection And Reporting

### PR DESCRIPTION
**Pull request by RealLucy**
This enhancement is intended to improve the hardware feature detection and reporting, in particular for more recently introduced hardware. The enhancement is a prerequisite for possible future feature exploitation. 

Reviews are highly welcome and appreciated.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8264173](https://bugs.openjdk.java.net/browse/JDK-8264173): [s390] Improve Hardware Feature Detection And Reporting

### Reviewers
 * [Martin Doerr](https://openjdk.java.net/census#mdoerr) (@TheRealMDoerr - **Reviewer**) ⚠️ Review applies to d15d115722dd298554a5d4907fd278fbcb046ae8
 * [Goetz Lindenmaier](https://openjdk.java.net/census#goetz) (@GoeLin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3196/head:pull/3196` \
`$ git checkout pull/3196`

Update a local copy of the PR: \
`$ git checkout pull/3196` \
`$ git pull https://git.openjdk.java.net/jdk pull/3196/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3196`

View PR using the GUI difftool: \
`$ git pr show -t 3196`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3196.diff">https://git.openjdk.java.net/jdk/pull/3196.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/cpu/s390/vm_version_s390.cpp [hunk 1]
```diff
@@ -61,18 +69,29 @@ unsigned int  VM_Version::_Icache_lineSize                          = DEFAULT_CA
...
+                                  };
 
 void VM_Version::initialize() {
   determine_features();      // Get processor capabilities.
   set_features_string();     // Set a descriptive feature indication.
 
-  if (Verbose) {
-    print_features();
+  if (Verbose || PrintAssembly || PrintStubCode) {
+    print_features_internal("CPU Version as detected internally:  ", PrintAssembly || PrintStubCode);
```
Comment (TheRealMDoerr) [inline src/hotspot/cpu/s390/vm_version_s390.cpp:94]: 2 spaces while other usages have no spaces?

**Review by TheRealMDoerr — APPROVED** [review]
I didn't expect the change to become that large. But it looks good to me. The lengthy output only gets generated with -XX:+Verbose. That's fine.

File: src/hotspot/cpu/s390/vm_version_s390.cpp (hunk 1 above)
Comment (RealLucy) [reply src/hotspot/cpu/s390/vm_version_s390.cpp:94]: You are right. These spaces were excessive. Now they are gone.

**Comment by RealLucy** [comment]
Thank you for the review, Martin!

**Comment by GoeLin** [comment]
LGTM

**Review by GoeLin — APPROVED** [review]

**Comment by RealLucy** [comment]
Thank you Goetz! 
I will now proceed and integrate the PR.

**Pull request closed without merging**